packages/codemods/src/transforms/styled-components-to-compiled/styled-components-to-compiled.tsx

```diff
@@ -53,40 +54,55 @@ const transformer = (fileInfo: FileInfo, api: API, options: Options): string =>
     allowedImportSpecifierNames: imports.styledComponentsSupportedImportNames,
   });
 
-  const taggedTemplateExpressionsWithAttrs = collection.find(
-    j.TaggedTemplateExpression,
-    ({ tag }: TaggedTemplateExpression) => {
-      if (tag.type === 'CallExpression') {
-        const { callee } = tag;
-
-        return (
-          callee.type === 'MemberExpression' &&
-          callee.object.type === 'MemberExpression' &&
-          callee.object.object.type === 'Identifier' &&
-          callee.object.object.name === 'styled' &&
-          callee.property.type === 'Identifier' &&
-          callee.property.name === 'attrs'
-        );
-      }
-
-      return false;
-    }
-  );
-
-  if (taggedTemplateExpressionsWithAttrs.length) {
-    convertStyledAttrsToComponent({
-      j,
-      plugins,
-      expressions: taggedTemplateExpressionsWithAttrs,
-    });
-  }
-
   applyVisitor({
     plugins,
     originalProgram,
     currentProgram: collection.find(j.Program).get(),
   });
 
+  const styledImport = collection.find(
```

Comment:
Can't we just put this logic in the default plugin `buildAttributes`, omit the local `styled` name logic there, and then in Jira hook into `buildAttributes` so that it renames it to the local name? If that doesn't work, we should find a way so that it does.

Reply:
we talked about it with @at-nathan and decided that in this particular case that makes sense, as we don't really want to transform any styled-component attrs usages but rather fix compiled incompatibilities. And in order to do that, we need to apply all transformations, including the program visitor, and only then transform attrs usages.